### PR TITLE
update operator default timeout

### PR DESCRIPTION
**Pull request by Vacant2333**
**What type of PR is this?**

**Which issue(s) this PR fixes**:
Fixes https://github.com/karmada-io/karmada/issues/3935

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: operator/pkg/karmadaresource/webhookconfiguration/webhookconfiguration.go [hunk 1]
```diff
@@ -12,8 +12,8 @@ import (
 	"github.com/karmada-io/karmada/operator/pkg/util/apiclient"
 )
 
-// EnsureWebhookconfiguration creates karmada webhook mutatingWebhookConfiguration and validatingWebhookConfiguration
-func EnsureWebhookconfiguration(client clientset.Interface, namespace, name, caBundle string) error {
+// EnsureWebhookConfiguration creates karmada webhook mutatingWebhookConfiguration and validatingWebhookConfiguration
```
Comment (RainbowMango) [inline operator/pkg/karmadaresource/webhookconfiguration/webhookconfiguration.go:15]: Nice finding!
If you find an issue like this, please feel free to send a PR.

Comment (Vacant2333) [reply operator/pkg/karmadaresource/webhookconfiguration/webhookconfiguration.go:15]: thanks,  In most cases, I tend to make these changes while submitting a pr

**Review by RainbowMango — COMMENTED** [review]
/lgtm

Leave approval to @calvin0327 for the final decision.

I see this PR updates the default timeout for APIserver and ETCD, what about the other components?

**Comment by Vacant2333** [comment]
> /lgtm
> 
> Leave approval to @calvin0327 for the final decision.
> 
> I see this PR updates the default timeout for APIserver and ETCD, what about the other components?

During my testing, I found that there was insufficient time to make the changes in these specific areas, while the others were fine.

**Comment by RainbowMango** [comment]
> During my testing, I found that there was insufficient time to make the changes in these specific areas, while the others were fine

It'd be great if we could keep all components with the same timeout.

**Comment by Vacant2333** [comment]
image

I updated the timeout duration on lines 74, 75, and 77 from 30 seconds to 120 seconds. As for the other components, they are all covered on lines 80 and 81, and their default timeout is already set to 120 seconds, so there's no need for updates there. 
Alright, I'll make the modifications accordingly.

File: operator/pkg/tasks/init/wait.go [hunk 2]
```diff
@@ -14,6 +14,8 @@ import (
 )
 
 var (
+	componentBeReadyTimeout = 120 * time.Second
```
Comment (calvin0327) [inline operator/pkg/tasks/init/wait.go:17]: Could you explain why it's 120 seconds?

Comment (Vacant2333) [reply operator/pkg/tasks/init/wait.go:17]: By default, they are both set to 30 seconds. Many times, if an image needs to be pulled, it's not sufficient. In my environment, it probably needs to be set to 60 seconds. Of course, I cannot provide an exact timeout, but I believe 120 seconds should be sufficient for use in the vast majority of scenarios.

Comment (calvin0327) [reply operator/pkg/tasks/init/wait.go:17]: Could you palease add some comments for the variable `componentBeReadyTimeout` ?

Comment (Vacant2333) [reply operator/pkg/tasks/init/wait.go:19]: ok

**Comment by XiShanYongYe-Chang** [comment]
Ask an again review for @calvin0327 @RainbowMango

**Review by RainbowMango — COMMENTED** [review]
/lgtm

Leave approval to @carlory for final decisions.

**Comment by carlory** [comment]
It makes senses to me. 

/approve

A better way is to retry the reconcile func if an error occur. but it's a bit complicated to do now.

**Pull request merged**